src/SIM/SIMoutput.C

```diff
@@ -1110,7 +1110,82 @@ bool SIMoutput::writeGlvE (const Vector& vec, int iStep, int& nBlock,
 void SIMoutput::closeGlv ()
 {
   if (myVtf) delete myVtf;
-  myVtf = 0;
+  myVtf = nullptr;
+}
+
+
+bool SIMoutput::dumpMatlabGrid (std::ostream& os, const std::string& name,
+                                const std::vector<std::string>& sets,
+                                double scale) const
+{
+  // Write function definition
+  os <<"function [Node, Element";
+  for (const std::string& setname : sets) os <<", "<< setname;
+  os <<"]="<< name << std::endl;
+
+  // Write out nodes for the specified topology sets
+  ASMbase* pch;
+  TopologySet::const_iterator tit;
+  for (const std::string& setname : sets)
+  {
+    std::set<int> nodeSet;
+    if ((tit = myEntitys.find(setname)) != myEntitys.end())
+      for (const TopItem& top : tit->second)
+        if ((pch = this->getPatch(top.patch)))
+          if (top.idim+1 == pch->getNoParamDim())
+          {
+            IntVec nodes;
+            pch->getBoundaryNodes(top.item,nodes);
+            for (int n : nodes) nodeSet.insert(n);
+          }
+
+    os <<"\n  "<< setname <<"=[";
+    int count = 0;
+    for (int node : nodeSet)
+      if (++count == 1)
+        os << node;
+      else if (count%16 == 1)
+        os <<",...\n    "<< node;
+      else
+        os <<", "<< node;
+    os <<"];"<< std::endl;
+  }
+
+  // Write out all nodes
+  os <<"\n  Node=[1";
+  size_t nnod = mySam->getNoNodes();
```

Comment:
On second thought, the element dump will in any case be invalid in case of mixed here, so instead I added an error return if meshes with more that 4 element nodes are encountered, and only 2D models.